lib/rules/no-observers.js

```diff
@@ -30,40 +31,67 @@ module.exports = {
       context.report(node, ERROR_MESSAGE);
     };
 
+    let importedEmberName = undefined;
+    let importedObserverName = undefined;
+    let importedObservesName = undefined;
+    let importedAddObserverName = undefined;
+
     return {
       CallExpression(node) {
-        if (ember.isModule(node, 'observer')) {
-          report(node);
-        }
-
         const { callee } = node;
-
         if (
-          (types.isThisExpression(callee.object) || types.isMemberExpression(callee)) &&
+          // observer();
+          types.isIdentifier(callee) &&
+          callee.name === importedObserverName
+        ) {
+          report(node);
+        } else if (
+          // Ember.observer();
+          types.isMemberExpression(callee) &&
+          types.isIdentifier(callee.object) &&
+          callee.object.name === importedEmberName &&
+          types.isIdentifier(callee.property) &&
+          callee.property.name === 'observer'
+        ) {
+          report(node);
+        } else if (
+          // this.addObserver();
+          // Ember.addObserver();
+          // foo.addObserver();
+          types.isMemberExpression(callee) &&
```

Comment:
There used to be a `types.isThisExpression(callee.object)` check here. Was it not necessary?

Reply:
In this block, we are checking for `ANYTHING.addObserver()`, so that's why I'm not checking specifically for `this` anymore.